devo/sender/data.py

```diff
@@ -625,3 +644,155 @@ def emit(self, record):
+
+        :param config -> Configuration object.
+        :return: Boolean true or raises an exception
+        """
+        try:
+            if not Path(config.key).is_file():
+                raise DevoSenderException(
+                    "Error in the configuration, "
+                    + config.key + " path does not exist")
+        except IOError as message:
+            if message.errno == errno.EACCES:
+                raise DevoSenderException(
+                    "Error in the configuration, "
+                    + config.key + " can't be read")
+        try:
+            if not Path(config.cert).is_file():
+                raise DevoSenderException(
+                    "Error in the configuration, "
+                    + config.cert + " path does not exist")
+        except IOError as message:
+            if message.errno == errno.EACCES:
+                raise DevoSenderException(
+                    "Error in the configuration, "
+                    + config.cert + " can't be read")
+        try:
+            if not Path(config.chain).is_file():
+                raise DevoSenderException(
+                    "Error in the configuration, "
+                    + config.chain + " path does not exist")
+        except IOError as message:
+            if message.errno == errno.EACCES:
+                raise DevoSenderException(
+                    "Error in the configuration, "
+                    + config.chain + " can't be read")
+        return True
+
+    @staticmethod
+    def check_config_certificate_key(config):
+        """
+        Check if both the certificate and the key
+        in the configuration are compatible with each other.
+
+        :param config -> Configuration object.
+        :return: Boolean true or raises an exception
+        """
+        certificate_raw = open(config.cert, "r").read()
+        key_raw = open(config.key, "r").read()
+        certificate_obj = crypto.load_certificate(
+            crypto.FILETYPE_PEM, certificate_raw)
+        private_key_obj = crypto.load_privatekey(crypto.FILETYPE_PEM, key_raw)
+        context = SSL.Context(SSL.TLSv1_METHOD)
+        context.use_privatekey(private_key_obj)
+        context.use_certificate(certificate_obj)
+        try:
+            context.check_privatekey()
+        except SSL.Error:
+            raise DevoSenderException(
+                "Error in the configuration, the key: " + config.key +
+                " is not compatible with the cert: " + config.cert)
+        return True
+
+    @staticmethod
+    def check_config_certificate_chain(config):
+        """
+        Check if both the certificate and the chain
+        in the configuration are compatible with each other.
+
+        :param config -> Configuration object.
+        :return: Boolean true or raises an exception
+        """
+        try:
+            certificate_raw = open(config.cert, "rb").read()
+            chain_raw = open(config.chain, "rb").read()
```

Comment:
No reason just remnants from all the test, in the next commit the should all be `"rb"`since the method expect bytes and not `str`, it will work with either but no need for an extra warning.